### DISCUSSION TITLE
[v5] - MSSQL preload not returning data

### DISCUSSION THREAD

**Discussion by anthrogan** [adonisjs/core#2387] [Help] [answered]
I can't for the life of my fathom what I'm doing wrong here.

Query:
![image](https://user-images.githubusercontent.com/30973884/111194535-3765f100-85b3-11eb-89f2-ddbb88e75c6c.png)

Sale relation in the Meter model:
![image](https://user-images.githubusercontent.com/30973884/111194604-45b40d00-85b3-11eb-9643-35c750c28135.png)

SQL Query (which works and returns data fine in SSMS)
![image](https://user-images.githubusercontent.com/30973884/111194723-60868180-85b3-11eb-9429-9268e2ea437d.png)

However the data returned has no sale prop:
![image](https://user-images.githubusercontent.com/30973884/111194828-7d22b980-85b3-11eb-9374-f3ba6a8a6f1a.png)

...Stumped. Am I doing something wrong? I can write this as a rawQuery but I kinda feel like I'm missing the point of model relations etc.

**Comment by aayush123** [comment]
Try logging the output of meter.$getRelated('sale'). Does it have the data you're looking for?

**Reply by anthrogan** [reply]
`undefiend`

**Comment by anthrogan** [comment, marked as answer]
This appears to be specifically an issue with mssql connections. I switched the models to look at a version of the data in a PG engine and it works as expected. I believe this related to mssql parameter syntax. However it works absolutely fine when doing something like `/sale/:id` The PG engine is what we are moving to, so its a few days behind in data. So if it exists in PG it has to exist in MSSQL

Working with PG -
![image](https://user-images.githubusercontent.com/30973884/111323988-435cbc00-8662-11eb-9ef3-d08039baa1c1.png)

![image](https://user-images.githubusercontent.com/30973884/111323883-2b853800-8662-11eb-87aa-730e2757d5b1.png)

Not working with MSSQL - 
![image](https://user-images.githubusercontent.com/30973884/111325456-9f741000-8663-11eb-8475-7d7518df4a35.png)

**Reply by thetutlage** [reply]
Can you share the migration files for both the tables. Looks like an issue with the data type mis-match between the foriegn key and the primary key

**Reply by anthrogan** [reply]
I just replicated the tables with migrations (the orig tables were not made from migrations as its a pre existing db) and it worked as expected, this is likely as you say it is a data mis-match somewhere db side. Not an adonis issue.

Thanks for the help 👍
